enterprise/models.py

```diff
@@ -746,9 +746,12 @@ def enroll(self, course_run_id, mode, cohort=None):
                 'is_upgrading': is_upgrading,
             })
             try:
-                EnterpriseCourseEnrollment.objects.get_or_create(
+                enrollment, created = EnterpriseCourseEnrollment.objects.update_or_create(
```

Comment:
@brittneyexline Do you think this will be an issue? Changing this from a `get_or_create` to an `update_or_create`?

This change was triggered because it looks like the Enterprise Course Enrollment is already created by the time we get to this point while converting a PendingEnrollment to an EnterpriseCourseEnrollment.

The outcome from this would be an updated Source if the `EnterpriseCourseEnrollment` already existed.

Reply:
If we're updating, could this overwrite the source?

Reply:
Yes, an update will change the source to what is set here.  I am trying to find where it is being created first right now.

Reply:
It looks like the issue is that we have a Task that is run on every save to the enrollment table that checks if the enrollment is an EnterpriseCourseEnrollment.  Then it creates the Enrollment.

In this case, we enroll the learner and then we create the EnterpriseCourseEnrollment in the task.  Then in the code, we attempt to `get_or_create` the EnterpriseCourseEnrollment with the Source and it already exists so the Source is not set.